cmd/replicate-add.go

```diff
@@ -167,21 +167,19 @@ func mainReplicateAdd(cliCtx *cli.Context) error {
 	deleteReplicationStatus := disableStatus
 	replicaSync := enableStatus
 	existingReplicationStatus := disableStatus
-	if cliCtx.IsSet("replicate") {
-		replSlice := strings.Split(cliCtx.String("replicate"), ",")
-		for _, opt := range replSlice {
-			switch strings.TrimSpace(strings.ToLower(opt)) {
-			case "delete-marker":
-				dmReplicateStatus = enableStatus
-			case "delete":
-				deleteReplicationStatus = enableStatus
-			case "metadata-sync", "replica-metadata-sync":
-				replicaSync = enableStatus
-			case "existing-objects":
-				existingReplicationStatus = enableStatus
-			default:
-				fatalIf(probe.NewError(fmt.Errorf("invalid value for --replicate flag %s", cliCtx.String("replicate"))), `--replicate flag takes one or more comma separated string with values "delete", "delete-marker", "metadata-sync", "existing-objects" or "" to disable these settings`)
-			}
+	replSlice := strings.Split(cliCtx.String("replicate"), ",")
```

Comment:
This means that --replicate will be always mandatory to pass in the cli, is that expected?

Reply:
if --replicate is not passed, it will use the defaults.

Reply:
> if --replicate is not passed, it will use the defaults.

Okay, but the new change breaks this behavior. replSlice slice will be always non empty here and it will fail with a fatal() in the next for loop